nautobot/extras/migrations/0057_collect_roles_from_related_apps_roles.py

```diff
@@ -61,22 +61,46 @@ def bulk_create_roles(apps, roles_to_create, content_types):
     """Bulk create role and set its contenttypes"""
     Role = apps.get_model("extras", "Role")
     ContentType = apps.get_model("contenttypes", "ContentType")
+    ObjectChange = apps.get_model("extras", "ObjectChange")
+
+    old_role_ct = None
+    new_role_ct = None
+
+    if hasattr(roles_to_create, "model"):
+        old_role_ct = ContentType.objects.get_for_model(roles_to_create.model)
+        # No sense fetching this if we don't have an old role content type
+        new_role_ct = ContentType.objects.get_for_model(Role)
 
     # Exclude roles whose names are already exists from bulk create.
     existing_roles = Role.objects.values_list("name", flat=True)
     roles_instances = [
-        Role(
-            name=data.name,
-            slug=data.slug,
-            description=data.description,
-            color=getattr(data, "color", color_map["default"]),
-            weight=getattr(data, "weight", None),
+        (
+            data,
+            Role(
+                name=data.name,
+                slug=data.slug,
+                description=data.description,
+                color=getattr(data, "color", color_map["default"]),
+                weight=getattr(data, "weight", None),
+            ),
```

Comment:
Should we instead be using `Role.objects.get_or_create()` as a way to handle existing roles?

Reply:
Maybe? But there's issues in the data migrations in the soon-to-be-deleted roles and associated objects.

We need some logic around "if slug matches but name differs" or "if name matches but description differs" etc.